src/orchestrator/errors.rs

```diff
@@ -38,6 +39,55 @@ pub enum Error {
     Cancelled,
 }
 
+impl Error {
+    /// Helper function that logs an error level trace event if the given client error is internal.
+    fn handle_internal_client_error(error: &clients::Error, from_client: String) {
+        debug!(?error, "error received from {}", from_client);
+        if matches!(error, clients::Error::Internal { .. }) {
+            error!("internal client error: {}", error);
+        }
+    }
```

Comment:
All of these error helper functions feel unnecessary IMO and out of scope for this PR.

Reply:
All other points addressed directly. For this one, I simplified to a singular function. The (now called) `handle_client_error` function exists so the orchestrator can transform `clients::Error` appropriately everywhere where it is returned, in particular:
- any errors related to some request are turned into HTTP errors to propagate to the request sender, where the internal error details are omitted
- any internal client errors are immediately logged when received by the orchestrator before error is turned into HTTP variant

Reply:
@pscoro I'm still not a fan of this; IMO maintaining an additional enum for each client type just for error handling seems unnecessary. Thinking about this more, I have some ideas to improve+simplify our errors (and I'm sure you do too), e.g. I think we can drop the "<Client>RequestFailed" variants and have a single `Client` variant encapsulating sufficient detail/context, etc.

As error handling enhancements are beyond the scope of this PR, can we drop these changes and address in a separate PR after we discuss? It appears they aren't required for the hyper migration unless I'm missing something.

Reply:
@declark1 fully understandable, I'll try to trim out all the error changes from this PR today and we can look into this further as a separate item. I kinda already figured the change I pushed yesterday wasn't ideal and would cause more discussion, the only requirements I wanted to preserve here in this PR is that error into client response logic is kept correct and that internal error messages (about impl details) dont end up in http error responses.

> IMO maintaining an additional enum for each client type just for error handling seems unnecessary. Thinking about this more, I have some ideas to improve+simplify our errors (and I'm sure you do too), e.g. I think we can drop the <Name>RequestFailed variants and have a single Client variant encapsulating sufficient detail/context, etc.

Yep this is the start of a larger discussion for sure (we should book time to discuss), fully agree that client type/kind shouldnt exist only for error handling, and whether it should exist is part of a larger design discussion. I didnt feel like this is an appropriate place to introduce it but just wanted to tie up the work I was doing yesterday and push an update even though I hadnt thought of a simpler alternative yet